pkg/networkservice/common/refresh/client.go

```diff
@@ -78,23 +89,31 @@ func (t *refreshClient) Close(ctx context.Context, conn *networkservice.Connecti
 			timer.Stop()
 			delete(t.connectionTimers, conn.GetId())
 		}
+		if canceller, ok := t.timerContextCancellers[conn.GetId()]; ok {
+			canceller()
+			delete(t.timerContextCancellers, conn.GetId())
```

Comment:
Should we also delete the entry from the map connectionTimers?

Reply:
It is deleted two lines above